pkg/apis/workspaces/v1alpha2/component_container.go

```diff
@@ -7,6 +7,21 @@ type ContainerComponent struct {
 	Endpoints     []Endpoint `json:"endpoints,omitempty" patchStrategy:"merge" patchMergeKey:"name"`
 }
 
+// Annotation specifies the annotations to be added to specific resources
+type Annotation struct {
+	// +optional
+	// Annotations to be added to deployment
+	Deployment map[string]string `json:"deployment,omitempty" patchStrategy:"merge" patchMergeKey:"name"`
+
+	// +optional
+	// Annotations to be added to service
+	Service map[string]string `json:"service,omitempty" patchStrategy:"merge" patchMergeKey:"name"`
+
+	// +optional
+	// Annotations to be added to Kubernetes Ingress or Openshift Route
+	Ingress map[string]string `json:"ingress,omitempty" patchStrategy:"merge" patchMergeKey:"name"`
```

Comment:
good point. I pushed a new commit to move the ingress/route annotation to be set under endpoint. so each endpoint can set it's own annotation. 
for Odo, endpoints in a pod share the same service, unless `dedicatedPod = true`. Any usecase in che will have multiple services for containers within same pod?

Reply:
> Any usecase in che will have multiple services for containers within same pod?

Currently no, but the spec doesn't require one service, so it could cause issues down the line. I think it still makes sense to keep service annotations workspace wide for now, though.